examples/utils/include/traccc/examples/utils/caching_memory_resource.hpp

```diff
@@ -0,0 +1,31 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023-2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+#include <vecmem/memory/binary_page_memory_resource.hpp>
+#include <vecmem/memory/memory_resource.hpp>
+
+namespace traccc {
+class caching_memory_resource final : public vecmem::memory_resource {
```

Comment:
I like the idea, but I would not like to have this here. Under the examples in this repository.

I would propose to teach `vecmem::binary_page_memory_resource` about this threshold itself. :thinking: Similar to:

https://github.com/acts-project/vecmem/blob/main/core/include/vecmem/memory/pool_memory_resource.hpp#L75

Or... we may not need to do anything actually... :thinking: Instead of creating a new memory resource class from scratch, you should just use:

https://github.com/acts-project/vecmem/blob/main/core/include/vecmem/memory/conditional_memory_resource.hpp

That's the thing that **you** wrote it for years ago already after all.

Reply:
Actually, the class to be used should be:

https://github.com/acts-project/vecmem/blob/main/core/include/vecmem/memory/choice_memory_resource.hpp